### ISSUE TITLE
any function to modify ICMP Packet code field?

### ISSUE THREAD

**Issue by Kilerd** [meh/rust-packet#10]
I cant find any function to modify it.

**Comment by meh** [comment]
At the moment you can't as changing the code changes the type of packet entirely, do you have a use-case in mind so I can see how I can try and support it?

**Comment by Kilerd** [comment]
in fact, i just wanna build a customized icmp packet. currently the api only supports three types of icmp: echo, information, and timestamp. but based on the RFC, exclude those were deprecated, builder should be built into 
- 0 – Echo Reply
- 3 – Destination Unreachable
- 5 – Redirect Message
- 8 – Echo Request
- 9 – Router Advertisement
- 10 – Router Solicitation
- 11 – Time Exceeded
- 12 – Parameter Problem: Bad IP header
- 13 – Timestamp
- 14 – Timestamp Reply
- 42 – Extended Echo Request
- 43 – Extended Echo Reply
- 253
- 254

most of them showed above cant be built through `Builder`. packet, as a common lib, i think should be more general, so adding `type(type: ICMPType)` and `code(code: u8)` into Builder would be better?

**Comment by meh** [comment]
Oh yeah, a lot of ICMP packet types are unimplemented, but the idea was to keep it type-safe and not allow you to build invalid ICMP packets.

Which packet types are you interested in right now? I'm also open to changing my mind on allowing building arbitrary ICMP packets, but it sort of goes against what I had in mind at the moment.

**Comment by Kilerd** [comment]
i agree that type-safe is quite important, but i think Builder should also have the unchecked method to create raw packet whose danger can be handled by user, just as `Packet::unchecked`. 
currently, what i am doing with `packet` in my project is using a specific ICMP packet as the information carrier between cluster nodes, this ICMP packet is a normal ICMP echo request and reply message witch a specific code(not zero).
so now i cant built the packet from `Builder`, and this's one of my "non-standard" usages.

combined with raw socket, i can handle those non-standard packets in my program.
the last, my point is that, maybe we can use a specific `Builder` for those common packet to keep the type-safe, like `icmp::echo::Builder`, `icmp::information::Builder`, and raw builder or kind of unchecked Builder should be provided to.
it's ok not to provide those raw builders, i can build it from lower network packet.
